internal/imports_test.go

```diff
@@ -0,0 +1,217 @@
+		"weather_forecast_sub/internal/handlers",
+	)
+}
+
+func Test_PkgSMPT_ShouldNotDependOn_Service(t *testing.T) {
+	t.Parallel()
+	archtest.Package(t, "weather_forecast_sub/pkg/email/smtp").ShouldNotDependOn(
+		"weather_forecast_sub/internal/service",
+	)
+}
+
+func Test_PkgSMPT_ShouldNotDependOn_Repository(t *testing.T) {
+	t.Parallel()
+	archtest.Package(t, "weather_forecast_sub/pkg/email/smtp").ShouldNotDependOn(
+		"weather_forecast_sub/internal/repository",
+	)
+}
+
+func Test_PkgErrors_ShouldNotDependOn_Handlers(t *testing.T) {
+	t.Parallel()
+	archtest.Package(t, "weather_forecast_sub/pkg/errors").ShouldNotDependOn(
+		"weather_forecast_sub/internal/handlers",
+	)
+}
+
+func Test_PkgErrors_ShouldNotDependOn_Service(t *testing.T) {
+	t.Parallel()
+	archtest.Package(t, "weather_forecast_sub/pkg/errors").ShouldNotDependOn(
+		"weather_forecast_sub/internal/service",
+	)
+}
+
+func Test_PkgErrors_ShouldNotDependOn_Repository(t *testing.T) {
+	t.Parallel()
+	archtest.Package(t, "weather_forecast_sub/pkg/errors").ShouldNotDependOn(
+		"weather_forecast_sub/internal/repository",
+	)
+}
+
+func Test_PkgHash_ShouldNotDependOn_Handlers(t *testing.T) {
+	t.Parallel()
+	archtest.Package(t, "weather_forecast_sub/pkg/hash").ShouldNotDependOn(
+		"weather_forecast_sub/internal/handlers",
+	)
+}
+
+func Test_PkgHash_ShouldNotDependOn_Service(t *testing.T) {
+	t.Parallel()
+	archtest.Package(t, "weather_forecast_sub/pkg/hash").ShouldNotDependOn(
+		"weather_forecast_sub/internal/service",
+	)
+}
+
+func Test_PkgHash_ShouldNotDependOn_Repository(t *testing.T) {
+	t.Parallel()
+	archtest.Package(t, "weather_forecast_sub/pkg/hash").ShouldNotDependOn(
+		"weather_forecast_sub/internal/repository",
+	)
+}
+
+func Test_PkgLogger_ShouldNotDependOn_Handlers(t *testing.T) {
+	t.Parallel()
+	archtest.Package(t, "weather_forecast_sub/pkg/logger").ShouldNotDependOn(
+		"weather_forecast_sub/internal/handlers",
+	)
+}
+
+func Test_PkgLogger_ShouldNotDependOn_Service(t *testing.T) {
+	t.Parallel()
+	archtest.Package(t, "weather_forecast_sub/pkg/logger").ShouldNotDependOn(
+		"weather_forecast_sub/internal/service",
+	)
+}
+
+func Test_PkgLogger_ShouldNotDependOn_Repository(t *testing.T) {
+	t.Parallel()
+	archtest.Package(t, "weather_forecast_sub/pkg/logger").ShouldNotDependOn(
+		"weather_forecast_sub/internal/repository",
+	)
+}
+
+func Test_PkgMigrations_ShouldNotDependOn_Handlers(t *testing.T) {
+	t.Parallel()
+	archtest.Package(t, "weather_forecast_sub/pkg/migrations").ShouldNotDependOn(
+		"weather_forecast_sub/internal/handlers",
+	)
+}
+
+func Test_PkgMigrations_ShouldNotDependOn_Service(t *testing.T) {
+	t.Parallel()
+	archtest.Package(t, "weather_forecast_sub/pkg/migrations").ShouldNotDependOn(
+		"weather_forecast_sub/internal/service",
+	)
+}
+
+func Test_PkgMigrations_ShouldNotDependOn_Repository(t *testing.T) {
+	t.Parallel()
+	archtest.Package(t, "weather_forecast_sub/pkg/migrations").ShouldNotDependOn(
```

Comment:
https://github.com/matthewmcnew/archtest/blob/master/archtest_test.go#L144
```suggestion
	archtest.Package(t, "weather_forecast_sub/pkg/...").ShouldNotDependOn(
```

Можемо ось так зробити, щоб не потрібно було для кожного підпакету робити окремий тест?

Reply:
Можемо, прикольно, я навіть не думав що так можна. Відрефачив.